src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,103 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [countOfSuccessSubmissions, setCountOfSuccessSubmissions] = useState(0);
+
+  // #region state
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgURL] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  //endregion
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    });
+
+    setTitle('');
+    setDescription('');
+    setImgURL('');
+    setImdbUrl('');
+    setImdbId('');
+
+    setCountOfSuccessSubmissions(countOfSuccessSubmissions + 1);
```

Comment:
```suggestion
    setCountOfSuccessSubmissions((prev) => prev + 1);
```
It is a good practice to update previous state like this